File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row = set()
        for r in board:  # r is a list
            row.clear()
            for num in r:
                if num in row and num != '.':
                    return False
                else:
                    row.add(num)

        col = set()
        for i in range(0, 9):
            col.clear()
            for j in range(0, 9):
                num = board[j][i]
                if num in col and num != '.':
                    return False
                else:
                    col.add(num)

        sets = [set() for i in range(9)]
        for i in range(0, 9):
            for j in range(0, 9):
                # max result for resi will be 6 because max result of (i/3) = 2 when i = 8
                resi = 3 * int(i / 3)

                resj = int(j / 3)
                num = board[j][i]
                if num in sets[resi+resj] and num != '.':
                    return False
                else:
                    sets[resi+resj].add(num)
        return True

File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function (board) {
    const set1 = new Set();
    const set2 = new Set();
    for (let i = 0; i < 9; i++) {
        for (let j = 0; j < 9; j++) {
            if (board[i][j] !== ".") {
                if (set1.has(board[i][j])) return false;
                set1.add(board[i][j]);
            }

            if (board[j][i] !== ".") {
                if (set2.has(board[j][i])) return false;
                set2.add(board[j][i]);
            }
        }
        set1.clear();
        set2.clear();
    }

    const set3 = new Set();
    let k = 1;
    for (let i = 0; i < 9; i += 3) {
        for (let j = 0; j < 9; j += 3) {
            for (let a = i; a < i + 3; a++) {
                for (let b = j; b < j + 3; b++) {
                    if (board[a][b] !== "." && set3.has(board[a][b])) return false;
                    set3.add(board[a][b]);
                }
            }
            set3.clear();
        }
    }
    return true;
};